x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.security.authc.jwt;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.support.ClaimSetting;
+import org.elasticsearch.xpack.core.security.authc.support.DelegatedAuthorizationSettings;
+import org.elasticsearch.xpack.core.ssl.SSLConfigurationSettings;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+public class JwtRealmSettings {
+
+    private JwtRealmSettings() {}
+
+    public static final String TYPE = "jwt";
+    private static final String CLIENT_AUTHENTICATION_PREFIX = RealmSettings.PREFIX + TYPE + ".client_authentication.";
+
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS = List.of(
+        "HS256",
+        "HS384",
+        "HS512",
+        "RS256",
+        "RS384",
+        "RS512",
+        "ES256",
+        "ES384",
+        "ES512",
+        "PS256",
+        "PS384",
+        "PS512"
+    );
+
+    public static final List<String> SUPPORTED_CLIENT_AUTHENTICATION_TYPE = List.of("sharedsecret", "ssl", "none");
+
+    // Default values and min/max constraints
+
+    private static final TimeValue DEFAULT_ALLOWED_CLOCK_SKEW = TimeValue.timeValueSeconds(60);
+    private static final List<String> DEFAULT_ALLOWED_SIGNATURE_ALGORITHMS = Collections.singletonList("RS256");
+    private static final List<String> DEFAULT_ALLOWED_AUDIENCES = Collections.emptyList();
```

Comment:
I don;t think we need this, we can use `List.of()` in the setting definition directly. "default allowed audiences" doesn't make much sense in the context of this realm

Reply:
My thinking was lists can be set to defaults of null or empty.

Explicitly stating an empty list removes any ambiguity. However, I can remove it. No preference one way or the other.

Reply:
I would like to discuss. I ran into some unexpected corner cases.

Reply:
Remove default empty list for allowed_audiences